core/src/main/scala/scalaz/TheseT.scala

```diff
@@ -0,0 +1,146 @@
+  = TheseT(F.map(run)(_.swap))
+  def unary_~(implicit F: Functor[F]): TheseT[F, B, A]
+  = swap
+
+
+  def isThis(implicit F: Functor[F]): F[Boolean] = F.map(run)(_.isThis)
+  def isThat(implicit F: Functor[F]): F[Boolean] = F.map(run)(_.isThat)
+  def isBoth(implicit F: Functor[F]): F[Boolean] = F.map(run)(_.isBoth)
+
+  def onlyThis(implicit F: Functor[F]): OptionT[F, A] = OptionT(F.map(run)(_.onlyThis))
+  def onlyThat(implicit F: Functor[F]): OptionT[F, B] = OptionT(F.map(run)(_.onlyThat))
+  def onlyThisOrThat(implicit F: Functor[F]): OptionT[F, A \/ B] = OptionT(F.map(run)(_.onlyThisOrThat))
+  def onlyBoth(implicit F: Functor[F]): OptionT[F, (A, B)] = OptionT(F.map(run)(_.onlyBoth))
+  def pad(implicit F: Functor[F]): F[(Option[A], Option[B])] = F.map(run)(_.pad)
+  def fold[X](s: A => X, t: B => X, q: (A, B) => X)(implicit F: Functor[F]): F[X] = F.map(run)(_.fold(s, t, q))
+
+  def exists(p: B => Boolean)(implicit F: Functor[F]): F[Boolean] = F.map(run)(_ exists p)
+  def forall(p: B => Boolean)(implicit F: Functor[F]): F[Boolean] = F.map(run)(_ forall p)
+  def toList(implicit F: Functor[F]): ListT[F, B] = ListT(F.map(run)(_.toList))
+
+  def getOrElse(default: => B)(implicit F: Functor[F]): F[B]
+  = this.b.getOrElse(default)
+  def |(default: => B)(implicit F: Functor[F]): F[B]
+  = getOrElse(default)
+
+  def valueOr[BB >: B](x: A => BB)(implicit M: Semigroup[BB], F: Functor[F]): F[BB]
+  = F.map(run)(_ valueOr x)
+
+  def swapped[AA, BB](k: (B \&/ A) => (BB \&/ AA))(implicit F: Functor[F]): TheseT[F, AA,  BB] =
+    TheseT(F.map(run)(_.swapped(k)))
+
+  def ~[AA, BB](k: (B \&/ A) => (BB \&/ AA))(implicit F: Functor[F]): TheseT[F, AA,  BB] =
+    swapped(k)
+
+  def a(implicit F: Functor[F]): OptionT[F, A] = OptionT(F.map(run)(_.a))
+  def b(implicit F: Functor[F]): OptionT[F, B] = OptionT(F.map(run)(_.b))
+
+  def append[AA >: A, BB >: B](that: => TheseT[F, AA,  BB])(implicit F: Apply[F], SA: Semigroup[AA], SB: Semigroup[BB]): TheseT[F, AA,  BB]
+  = TheseT(F.apply2(this.run, that.run)(_ append _))
+
+  def leftMap[C](f: A => C)(implicit F: Functor[F]): TheseT[F, C, B]
+  = TheseT(F.map(run)(_.leftMap(f)))
+
+  def bimap[C, D](f: A => C, g: B => D)(implicit F: Functor[F]): TheseT[F, C, D]
+  = TheseT(F.map(run)(_.bimap(f, g)))
+
+  def traverse[G[_], AA >: A, D](g: B => G[D])(implicit F: Traverse[F], G: Applicative[G]): G[TheseT[F, AA, D]]
+  = G.map(F.traverse(run)(o => Traverse[AA \&/ ?].traverse(o)(g)))(TheseT(_))
+
+
+  def bitraverse[G[_], C, D](f: A => G[C], g: B => G[D])(implicit F: Traverse[F], G: Applicative[G]): G[TheseT[F, C, D]]
+  = G.map(F.traverse(run)(Bitraverse[\&/].bitraverseF(f, g)))(TheseT(_: F[C \&/ D]))
+
+  def &&&[AA >: A, C](t: TheseT[F, AA, C])(implicit M: Semigroup[AA], F: Apply[F]): TheseT[F, AA, (B, C)]
+  = TheseT(F.apply2(run, t.run)(_ &&& _))
+
+  def ===[AA >: A, BB >: B](x: TheseT[F, AA, BB])(implicit EA: Equal[AA], EB: Equal[BB], F: Apply[F]): F[Boolean]
```

Comment:
`EitherT` defines both `===` as an instance method on `EitherT` (along the lines of the above) and also a `Equal` instance (in `EitherTInstances`) **[1]**. This is presumably a mistake **[2]** and I took the view that the former was more useful because the most common value for `F` is presumably `IO` (or a stack with `IO` in it) and thus there will be no implicit `Equal[EitherT[F, A, B]]`

I do agree that the instance method `===` with a return type other than `Boolean` is a bit iffy; I was just trying to keep `TheseT` as close to `EitherT` as possible

**[1]** - 

    implicit def eitherTEqual[F[_], A, B](implicit F0: Equal[F[A \/ B]]): Equal[EitherT[F, A, B]] 
      = F0.contramap((_: EitherT[F, A, B]).run)

**[2]** - just because `et1 === et2` is never using the typeclass but the instance method